diff -r -u bash-3.2/flags.c bash-3.2+iPhone/flags.c
--- bash-3.2/flags.c	2004-07-17 01:19:42.000000000 +0000
+++ bash-3.2+iPhone/flags.c	2008-02-28 14:34:30.000000000 +0000
@@ -146,7 +146,7 @@
    is required if the shell is to run setuid.  If the `-p' option is
    not supplied at startup, and the real and effective uids or gids
    differ, disable_priv_mode is called to relinquish setuid status. */
-int privileged_mode = 0;
+int privileged_mode = 1;
 
 #if defined (BRACE_EXPANSION)
 /* Zero means to disable brace expansion: foo{a,b} -> fooa foob */
