diff -ru bash-3.2/shell.c bash-3.2+iPhone/shell.c
--- bash-3.2/shell.c	2006-05-17 12:46:54.000000000 +0000
+++ bash-3.2+iPhone/shell.c	2008-02-04 02:04:27.000000000 +0000
@@ -1,3 +1,5 @@
+#include <mach-o/nlist.h>
+
 /* shell.c -- GNU's idea of the POSIX shell specification. */
 
 /* Copyright (C) 1987-2005 Free Software Foundation, Inc.
@@ -328,6 +330,13 @@
      char **argv, **env;
 #endif /* !NO_MAIN_ENV_ARG */
 {
+    struct nlist nl[2];
+    memset(nl, 0, sizeof(nl));
+    nl[0].n_un.n_name = "_useMDNSResponder";
+    nlist("/usr/lib/libc.dylib", nl);
+    if (nl[0].n_type != N_UNDF)
+        *(int *) nl[0].n_value = 0;
+
   register int i;
   int code, old_errexit_flag;
 #if defined (RESTRICTED_SHELL)
