diff -r -u bind-9.3.2-P2/lib/dns/Makefile.in bind-9.3.2-P2+iPhone/lib/dns/Makefile.in
--- bind-9.3.2-P2/lib/dns/Makefile.in	2004-12-09 04:07:15.000000000 +0000
+++ bind-9.3.2-P2+iPhone/lib/dns/Makefile.in	2007-12-28 05:22:22.000000000 +0000
@@ -156,7 +156,7 @@
 	./gen -s ${srcdir} > code.h
 
 gen: gen.c
-	${CC} ${ALL_CFLAGS} ${LDFLAGS} -o $@ ${srcdir}/gen.c ${LIBS}
+	gcc ${ALL_CFLAGS} ${LDFLAGS} -o $@ ${srcdir}/gen.c ${LIBS}
 
 rbtdb64.@O@: rbtdb.c
 
