diff -ru cydia-1.0.2712/uicaboodle.m/BrowserView.m cydia-1.0.2712+iPhone/uicaboodle.m/BrowserView.m
--- cydia-1.0.2712/uicaboodle.m/BrowserView.m	2009-01-26 10:31:46.000000000 +0000
+++ cydia-1.0.2712+iPhone/uicaboodle.m/BrowserView.m	2009-01-26 10:53:00.000000000 +0000
@@ -352,7 +352,7 @@
 }
 
 - (void) webView:(WebView *)sender runJavaScriptAlertPanelWithMessage:(NSString *)message initiatedByFrame:(WebFrame *)frame {
-    if ([self _allowJavaScriptPanel])
+    if (![self _allowJavaScriptPanel])
         return;
 
     UIActionSheet *sheet = [[[UIActionSheet alloc]
