diff -ru cydia-1.0.3028/cydia/Cydia.app/German.lproj/Localizable.strings cydia-1.0.3028+iPhone/cydia/Cydia.app/German.lproj/Localizable.strings
--- cydia-1.0.3028/cydia/Cydia.app/German.lproj/Localizable.strings	2009-09-08 02:17:13.000000000 +0000
+++ cydia-1.0.3028+iPhone/cydia/Cydia.app/German.lproj/Localizable.strings	2009-09-08 06:05:21.000000000 +0000
@@ -179,7 +179,7 @@
 "SPONSOR" = "Sponsor";
 "SPONSORING_PRODUCTS" = "Unterstützende Produkte";
 "STATISTICS" = "Statistik";
-"STILL_BROKEN_EX" = "Der Versuch, die kaputten Sektionen zu reparieren hat nichts gebracht."
+"STILL_BROKEN_EX" = "Der Versuch, die kaputten Sektionen zu reparieren hat nichts gebracht.";
 "STORAGE" = "Speicherplatz";
 "STORAGE_EX" = "Nachsehen, wieviel Speicherplatz für neue Pakete vorhanden ist.";
 "SUBMIT" = "Absenden";
