diff -ru cydia-1.0.2473/cydia/Library/startup cydia-1.0.2473+iPhone/cydia/Library/startup
--- cydia-1.0.2473/cydia/Library/startup	2008-07-13 12:18:54.000000000 +0000
+++ cydia-1.0.2473+iPhone/cydia/Library/startup	2008-07-17 17:10:17.000000000 +0000
@@ -1,6 +1,5 @@
 #!/bin/bash
 export PATH=$PATH:/usr/sbin:/usr/bin:/sbin:/bin
-sysctl -w security.mac.proc_enforce=0 security.mac.vnode_enforce=0 &>/dev/null
 /usr/libexec/cydia/firmware.sh
 debs=/var/root/Media/Cydia/AutoInstall/*.deb
 dpkg -i ${debs}
