diff -ru ruby-1.9.1-p429/configure.in ruby-1.9.1-p429+iPhone/configure.in
--- ruby-1.9.1-p429/configure.in	2010-05-03 03:06:07.000000000 +0000
+++ ruby-1.9.1-p429+iPhone/configure.in	2010-08-07 23:16:58.000000000 +0000
@@ -1418,7 +1418,7 @@
 	rhapsody*)	: ${LDSHARED='cc -dynamic -bundle -undefined suppress'}
 			: ${LDFLAGS=""}
 			rb_cv_dlopen=yes ;;
-	darwin*)	: ${LDSHARED='cc -dynamic -bundle -undefined suppress -flat_namespace'}
+	darwin*)	: ${LDSHARED='$(CC) -dynamic -bundle -undefined suppress -flat_namespace'}
 			: ${LDFLAGS=""}
 			: ${LIBPATHENV=DYLD_LIBRARY_PATH}
 			# /usr/local/include is always searched for
@@ -1578,15 +1578,15 @@
 
 case "$target_os" in
   linux* | gnu* | k*bsd*-gnu)
-	STRIP='strip -S -x';;
+	STRIP=${STRIP}' -S -x';;
   nextstep*)
-	STRIP='strip -A -n';;
+	STRIP=${STRIP}' -A -n';;
   openstep*)
-	STRIP='strip -A -n';;
+	STRIP=${STRIP}' -A -n';;
   rhapsody*)
-	STRIP='strip -A -n';;
+	STRIP=${STRIP}' -A -n';;
   darwin*)
-	STRIP='strip -A -n';;
+	STRIP=${STRIP}' -A -n';;
 esac
 
 EXTSTATIC=
@@ -1781,7 +1781,7 @@
 	;;
     darwin*)
 	LIBRUBY_SO='lib$(RUBY_SO_NAME).$(MAJOR).$(MINOR).$(TEENY).dylib'
-	LIBRUBY_LDSHARED='cc -dynamiclib -undefined suppress -flat_namespace'
+	LIBRUBY_LDSHARED='$(CC) -dynamiclib -undefined suppress -flat_namespace'
 	LIBRUBY_DLDFLAGS='-install_name $(libdir)/lib$(RUBY_SO_NAME).dylib -current_version $(MAJOR).$(MINOR).$(TEENY) -compatibility_version $(ruby_version)'
 	LIBRUBY_ALIASES='lib$(RUBY_SO_NAME).$(MAJOR).$(MINOR).dylib lib$(RUBY_SO_NAME).dylib'
 	;;
Only in ruby-1.9.1-p429+iPhone: configure.in.orig
Only in ruby-1.9.1-p429+iPhone: configure.in.rej
