diff -ru dpkg-1.14.25/lib/Makefile.am dpkg-1.14.25+iPhone/lib/Makefile.am
--- dpkg-1.14.25/lib/Makefile.am	2009-02-01 06:18:58.000000000 +0000
+++ dpkg-1.14.25+iPhone/lib/Makefile.am	2009-04-11 01:11:30.000000000 +0000
@@ -41,4 +41,5 @@
 	trigdeferred.l \
 	utils.c \
 	varbuf.c \
-	vercmp.c
+	vercmp.c \
+	obstack.c
diff -ru dpkg-1.14.25/lib/tarfn.c dpkg-1.14.25+iPhone/lib/tarfn.c
--- dpkg-1.14.25/lib/tarfn.c	2009-02-01 06:18:59.000000000 +0000
+++ dpkg-1.14.25+iPhone/lib/tarfn.c	2009-04-11 01:11:30.000000000 +0000
@@ -55,6 +55,14 @@
 	return n;
 }
 
+int strnlen(const char *s, int size) {
+    int i;
+    for (i = 0; i != size; ++i)
+        if (s[i] == '\0')
+            return i;
+    return size;
+}
+
 /* String block to C null-terminated string */
 static char *
 StoC(const char *s, int size)
diff -ru dpkg-1.14.25/m4/funcs.m4 dpkg-1.14.25+iPhone/m4/funcs.m4
--- dpkg-1.14.25/m4/funcs.m4	2009-01-25 01:19:12.000000000 +0000
+++ dpkg-1.14.25+iPhone/m4/funcs.m4	2009-04-11 01:11:30.000000000 +0000
@@ -5,7 +5,7 @@
 # Define HAVE_VA_COPY if we have va_copy, fail if they can't be assigned
 AC_DEFUN([DPKG_FUNC_VA_COPY],
 [AC_CACHE_CHECK([for va_copy], [dpkg_cv_va_copy],
-[AC_TRY_RUN(
+[AC_TRY_COMPILE(
 [#include <stdarg.h>
 main() {
 va_list v1, v2;
diff -ru dpkg-1.14.25/utils/start-stop-daemon.c dpkg-1.14.25+iPhone/utils/start-stop-daemon.c
--- dpkg-1.14.25/utils/start-stop-daemon.c	2009-02-01 06:19:02.000000000 +0000
+++ dpkg-1.14.25+iPhone/utils/start-stop-daemon.c	2009-04-11 01:13:30.000000000 +0000
@@ -34,7 +34,7 @@
 #  define OSOpenBSD
 #elif defined(hpux)
 #  define OShpux
-#elif defined(__FreeBSD__)
+#elif defined(__FreeBSD__) || 1
 #  define OSFreeBSD
 #elif defined(__NetBSD__)
 #  define OSNetBSD
