diff -ru gdb-962/src/gdb/configure.ac gdb-962+iPhone/src/gdb/configure.ac
--- gdb-962/src/gdb/configure.ac	2008-03-05 23:09:41.000000000 +0000
+++ gdb-962+iPhone/src/gdb/configure.ac	2008-11-11 05:45:29.000000000 +0000
@@ -1681,16 +1681,7 @@
    AC_CONFIG_SUBDIRS(multi-ice)
 fi
 
-# We only build gdbserver automatically if host and target are the same.
-if test "x$target" = "x$host"; then
-  AC_MSG_CHECKING(whether gdbserver is supported on this host)
-  if test "x$build_gdbserver" = xyes; then
-    AC_MSG_RESULT(yes)
-    AC_CONFIG_SUBDIRS(gdbserver)
-  else
-    AC_MSG_RESULT(no)
-  fi
-fi
+AC_CONFIG_SUBDIRS(gdbserver)
 
 # We build rdi-share on ARM-based targets, as instructed by configure.tgt.
 if test "x$build_rdi_share" = xyes; then
