diff -ru gdb-961/src/gdb/macosx/macosx-nat-inferior.c gdb-961+iPhone/src/gdb/macosx/macosx-nat-inferior.c
--- gdb-961/src/gdb/macosx/macosx-nat-inferior.c	2008-05-29 23:05:46.000000000 +0000
+++ gdb-961+iPhone/src/gdb/macosx/macosx-nat-inferior.c	2008-07-04 11:11:37.000000000 +0000
@@ -2189,7 +2189,7 @@
 	  error ("Couldn't set the binary preferences, error: %d", retval);
     }
   retval = posix_spawnattr_setpgroup (&attr, debug_setpgrp);
-  if (retval != 0 || copied != 1)
+  if (retval != 0)
     error ("Couldn't set the process group, error: %d", retval);
 
   retval = posix_spawnp (&new_pid, fileptr, NULL,  &attr, argv, env);
