diff -ru gdb-961/src/gdb/configure.ac gdb-961+iPhone/src/gdb/configure.ac
--- gdb-961/src/gdb/configure.ac	2008-03-05 23:09:41.000000000 +0000
+++ gdb-961+iPhone/src/gdb/configure.ac	2008-07-04 10:20:50.000000000 +0000
@@ -897,7 +897,7 @@
 
 # APPLE LOCAL: Use the pre-conformant headers/functions so we stand a 
 # better chance of running on older systems.
-UNIX03_UGH_CFLAGS=-D__DARWIN_UNIX03=0
+UNIX03_UGH_CFLAGS=-D__DARWIN_UNIX03=1
 AC_SUBST(UNIX03_UGH_CFLAGS)
 
 # APPLE LOCAL: Enable MIG checking for macosx/macosx.defs.
