diff -ru gdb-952/src/gdb/config/arm/macosx.mh gdb-952+iPhone/src/gdb/config/arm/macosx.mh
--- gdb-952/src/gdb/config/arm/macosx.mh	2008-03-01 01:24:09.000000000 +0000
+++ gdb-952+iPhone/src/gdb/config/arm/macosx.mh	2008-04-11 08:41:15.000000000 +0000
@@ -47,10 +47,10 @@
 
 CFM_CFLAGS = -DWITH_CFM=0
 
 _all: all
 
 excServer.c excUser.c: macosx.defs
-	$(NEXT_ROOT)/usr/bin/mig -I. $<
+	mig -I. $<
 
 clean::
 	rm -f excServer.c excUser.c exc.h
diff -ru gdb-952/src/gdb/config/i386/macosx.mh gdb-952+iPhone/src/gdb/config/i386/macosx.mh
--- gdb-952/src/gdb/config/i386/macosx.mh	2008-03-01 01:24:10.000000000 +0000
+++ gdb-952+iPhone/src/gdb/config/i386/macosx.mh	2008-04-11 08:40:40.000000000 +0000
@@ -56,7 +56,7 @@
 _all: all
 
 $(MACH_EXC_SERVER_PREFACE)excServer.c $(MACH_EXC_SERVER_PREFACE)excUser.c: macosx.defs
-	$(SDKROOT)/usr/bin/mig -I. $<
+	mig -I. $<
 
 clean::
 	rm -f $(MACH_EXC_SERVER_PREFACE)excServer.c $(MACH_EXC_SERVER_PREFACE)excUser.c $(MACH_EXC_SERVER_PREFACE)exc.h
diff -ru gdb-952/src/gdb/config/powerpc/macosx.mh gdb-952+iPhone/src/gdb/config/powerpc/macosx.mh
--- gdb-952/src/gdb/config/powerpc/macosx.mh	2008-03-01 01:24:10.000000000 +0000
+++ gdb-952+iPhone/src/gdb/config/powerpc/macosx.mh	2008-04-11 08:41:32.000000000 +0000
@@ -60,7 +60,7 @@
 _all: all
 
 $(MACH_EXC_SERVER_PREFACE)excServer.c $(MACH_EXC_SERVER_PREFACE)excUser.c: macosx.defs
-	$(SDKROOT)/usr/bin/mig -I. $<
+	mig -I. $<
 
 clean::
 	rm -f $(MACH_EXC_SERVER_PREFACE)excServer.c $(MACH_EXC_SERVER_PREFACE)excUser.c $(MACH_EXC_SERVER_PREFACE)exc.h
