diff -ru gdb-952/src/gdb/config/arm/macosx.mh gdb-952+iPhone/src/gdb/config/arm/macosx.mh
--- gdb-952/src/gdb/config/arm/macosx.mh	2008-03-01 01:24:09.000000000 +0000
+++ gdb-952+iPhone/src/gdb/config/arm/macosx.mh	2008-04-11 07:38:06.000000000 +0000
@@ -43,7 +43,7 @@
 
 MH_CFLAGS = -I$(srcdir)/macosx
 
-MH_CFLAGS += -DUSE_PTHREADS=1 -Wall -Wimplicit -Wno-long-double
+MH_CFLAGS += -DUSE_PTHREADS=1 -Wall -Wimplicit
 
 CFM_CFLAGS = -DWITH_CFM=0
 
