diff -r -ru mobileterminal-286/Log.h mobileterminal-286+iPhone/Log.h
--- mobileterminal-286/Log.h	2008-05-07 08:33:09.000000000 +0000
+++ mobileterminal-286+iPhone/Log.h	2008-05-07 08:15:04.000000000 +0000
@@ -1,6 +1,7 @@
 
 #include <Foundation/Foundation.h>
 #include <CoreFoundation/CoreFoundation.h>
+#import <CoreGraphics/CGGeometry.h>
 
 #define logf(s,...)    [FileLog logFile:__FILE__ lineNumber:__LINE__ format:(s),##__VA_ARGS__]
 #define logfRect(s, r) [FileLog logFile:__FILE__ lineNumber:__LINE__ string:(s) rect:(r)] 
diff -r -ru mobileterminal-286/Preferences.m mobileterminal-286+iPhone/Preferences.m
--- mobileterminal-286/Preferences.m	2008-05-07 08:33:09.000000000 +0000
+++ mobileterminal-286+iPhone/Preferences.m	2008-05-07 08:22:11.000000000 +0000
@@ -13,7 +13,7 @@
 #import "Log.h"
 
 #import <UIKit/UISimpleTableCell.h> 
-#import "UIFieldEditor.h"
+#import <UIKit/UIFieldEditor.h>
 
 //_______________________________________________________________________________
 //_______________________________________________________________________________
diff -r -ru mobileterminal-286/Tools.h mobileterminal-286+iPhone/Tools.h
--- mobileterminal-286/Tools.h	2008-05-07 08:33:09.000000000 +0000
+++ mobileterminal-286+iPhone/Tools.h	2008-05-07 08:19:04.000000000 +0000
@@ -3,6 +3,7 @@
 //  Terminal
 
 #import <Foundation/Foundation.h>
+#import <CoreGraphics/CGImage.h>
 
 //_______________________________________________________________________________
 
diff -r -ru mobileterminal-286/Tools.m mobileterminal-286+iPhone/Tools.m
--- mobileterminal-286/Tools.m	2008-05-07 08:33:09.000000000 +0000
+++ mobileterminal-286+iPhone/Tools.m	2008-05-07 08:21:40.000000000 +0000
@@ -4,6 +4,7 @@
 
 #import "Tools.h"
 #import "Log.h"
+#import <ImageIO/CGImageDestination.h>
 
 //_______________________________________________________________________________
 //_______________________________________________________________________________
