diff -ru mobileterminal-273/Makefile mobileterminal-273+iPhone/Makefile
--- mobileterminal-273/Makefile	2007-10-01 08:23:11.000000000 +0000
+++ mobileterminal-273+iPhone/Makefile	2008-04-09 12:22:02.000000000 +0000
@@ -1,5 +1,5 @@
 CC=arm-apple-darwin-gcc
-CFLAGS=-Wall -Werror -O7
+CFLAGS=-Wall -O7
 LDFLAGS=-lobjc -framework CoreFoundation -framework Foundation \
         -framework UIKit -framework LayerKit -framework CoreGraphics \
         -framework GraphicsServices -lcurses
