diff -ru class-dump-3.1.2-x/src/CDObjCSegmentProcessor-Private.m class-dump-3.1.2-x+iPhone/src/CDObjCSegmentProcessor-Private.m
--- class-dump-3.1.2-x/src/CDObjCSegmentProcessor-Private.m	2008-04-20 04:27:18.000000000 +0000
+++ class-dump-3.1.2-x+iPhone/src/CDObjCSegmentProcessor-Private.m	2008-09-02 16:27:14.000000000 +0000
@@ -467,7 +467,9 @@
     
     // Search super class in Symtab first. We just need a super class name only so we don't need to get
     // detail information about super
-    NSString * superName = [NSString stringWithString:[[machOFile symbolTable] findSymbolByValue:objcClass.super_class]];
+    NSString * superName = [[machOFile symbolTable] findSymbolByValue:objcClass.super_class];
+    if (superName != nil)
+       superName = [NSString stringWithString:superName];
     // ptr = [machOFile pointerFromVMAddr:objcClass.super_class];
     if ( superName != nil ) {
        if ( [superName hasPrefix:@"_OBJC_CLASS_$_"] ) 
@@ -928,7 +930,9 @@
         return nil;
 
     // class - We just need name
-    NSString * class_name = [NSString stringWithString:[[machOFile symbolTable] findSymbolByValue:objcCategory.class]];
+    NSString * class_name = [[machOFile symbolTable] findSymbolByValue:objcCategory.class];
+    if (class_name != nil)
+        class_name = [NSString stringWithString:class_name];
     if ( class_name != nil ) {
        if ( [class_name hasPrefix:@"_OBJC_CLASS_$_"] )
           class_name = [class_name substringFromIndex:14];
