diff -ru aptitude-0.4.4.orig/src/generic/util/eassert.cc aptitude-0.4.4.orig+iPhone/src/generic/util/eassert.cc
--- aptitude-0.4.4.orig/src/generic/util/eassert.cc	2006-10-27 04:26:08.000000000 +0000
+++ aptitude-0.4.4.orig+iPhone/src/generic/util/eassert.cc	2008-05-13 09:07:58.000000000 +0000
@@ -31,6 +31,6 @@
 
 std::string AssertionFailure::errmsg() const
 {
-  return ssprintf("%s:%d: %s: Assertion \"%s\" failed.",
+  return ssprintf("%s:%ld: %s: Assertion \"%s\" failed.",
 		  file.c_str(), line, func.c_str(), exp.c_str());
 }
diff -ru aptitude-0.4.4.orig/src/generic/util/util.cc aptitude-0.4.4.orig+iPhone/src/generic/util/util.cc
--- aptitude-0.4.4.orig/src/generic/util/util.cc	2006-10-27 04:26:08.000000000 +0000
+++ aptitude-0.4.4.orig+iPhone/src/generic/util/util.cc	2008-05-13 09:08:45.000000000 +0000
@@ -154,9 +154,9 @@
     {
       char *buf = new char[bufsize];
 
-      char *result = strerror_r(errnum, buf, bufsize);
+      int result = strerror_r(errnum, buf, bufsize);
 
-      if(result == NULL)
+      if(result == -1)
 	{
 	  delete[] buf;
 
diff -ru aptitude-0.4.4.orig/src/vscreen/transcode.cc aptitude-0.4.4.orig+iPhone/src/vscreen/transcode.cc
--- aptitude-0.4.4.orig/src/vscreen/transcode.cc	2006-10-27 04:26:08.000000000 +0000
+++ aptitude-0.4.4.orig+iPhone/src/vscreen/transcode.cc	2008-05-13 09:30:43.000000000 +0000
@@ -22,6 +22,7 @@
 #include <generic/util/eassert.h>
 #include <errno.h>
 #include <iconv.h>
+#include <langinfo.h>
 
 using namespace std;
 
