diff -ru jamvm-1.5.1/configure.ac jamvm-1.5.1+iPhone/configure.ac
--- jamvm-1.5.1/configure.ac	2007-12-16 04:05:21.000000000 +0000
+++ jamvm-1.5.1+iPhone/configure.ac	2008-03-01 05:16:44.000000000 +0000
@@ -238,10 +238,9 @@
         enable_zip=no
     fi])
 
-AC_CHECK_LIB(ffi,ffi_call,,
-    [if test "$enable_ffi" != no; then
-        AC_MSG_ERROR(cannot find libffi)
-    fi])
+if test "$enable_ffi" != no; then
+    AC_CHECK_LIB(ffi,ffi_call,,AC_MSG_ERROR(cannot find libffi))
+fi
 
 dnl Checks for header files.
 AC_HEADER_STDC
