diff -ru cydia-1.0.3032/cydia/Cydia.mm cydia-1.0.3032+iPhone/cydia/Cydia.mm
--- cydia-1.0.3032/cydia/Cydia.mm	2009-09-09 22:03:03.000000000 +0000
+++ cydia-1.0.3032+iPhone/cydia/Cydia.mm	2009-09-10 03:46:28.000000000 +0000
@@ -1932,6 +1932,11 @@
         CYString &rhi(PackageName(rhs, @selector(cyname)));
         CFStringRef lhn(lhi), rhn(rhi);
 
+        if (lhn == NULL)
+            return rhn == NULL ? NSOrderedSame : NSOrderedAscending;
+        else if (rhn == NULL)
+            return NSOrderedDescending;
+
         _profile(PackageNameCompare$NumbersLast)
             if (!lhi.empty() && !rhi.empty()) {
                 UniChar lhc(CFStringGetCharacterAtIndex(lhn, 0));
