diff -ru cydia-1.0.2595/cydia/Cydia.mm cydia-1.0.2595+iPhone/cydia/Cydia.mm
--- cydia-1.0.2595/cydia/Cydia.mm	2008-09-23 23:10:25.000000000 +0000
+++ cydia-1.0.2595+iPhone/cydia/Cydia.mm	2008-11-21 17:46:41.000000000 +0000
@@ -43,7 +43,6 @@
 #include <GraphicsServices/GraphicsServices.h>
 #include <Foundation/Foundation.h>
 
-#include <WebCore/DOMHTML.h>
 #import <QuartzCore/CALayer.h>
 
 #import <UIKit/UIKit.h>
@@ -51,6 +50,14 @@
 // XXX: remove
 #import <MessageUI/MailComposeController.h>
 
+#include <WebKit/DOMCSSPrimitiveValue.h>
+#include <WebKit/DOMCSSStyleDeclaration.h>
+#include <WebKit/DOMDocument.h>
+#include <WebKit/DOMHTMLAnchorElement.h>
+#include <WebKit/DOMHTMLBodyElement.h>
+#include <WebKit/DOMNodeList.h>
+#include <WebKit/DOMRGBColor.h>
+
 #include <WebKit/WebFrame.h>
 #include <WebKit/WebView.h>
 
@@ -181,27 +188,6 @@
 
 extern "C" int UIApplicationMain(int argc, char *argv[], NSString *principalClassName, NSString *delegateClassName);
 
-extern NSString *kUIButtonBarButtonAction;
-extern NSString *kUIButtonBarButtonInfo;
-extern NSString *kUIButtonBarButtonInfoOffset;
-extern NSString *kUIButtonBarButtonSelectedInfo;
-extern NSString *kUIButtonBarButtonStyle;
-extern NSString *kUIButtonBarButtonTag;
-extern NSString *kUIButtonBarButtonTarget;
-extern NSString *kUIButtonBarButtonTitle;
-extern NSString *kUIButtonBarButtonTitleVerticalHeight;
-extern NSString *kUIButtonBarButtonTitleWidth;
-extern NSString *kUIButtonBarButtonType;
-
-typedef enum {
-    kUIProgressIndicatorStyleLargeWhite = 0,
-    kUIProgressIndicatorStyleMediumWhite = 1,
-    kUIProgressIndicatorStyleMediumBrown = 2,
-    kUIProgressIndicatorStyleSmallWhite = 3,
-    kUIProgressIndicatorStyleSmallBlack = 4,
-    kUIProgressIndicatorStyleTinyWhite = 5,
-} UIProgressIndicatorStyle;
-
 typedef enum {
     kUIControlEventMouseDown = 1 << 0,
     kUIControlEventMouseMovedInside = 1 << 2, // mouse moved inside control target
@@ -4759,9 +4745,9 @@
 
         //NSNotificationCenter *center = [NSNotificationCenter defaultCenter];
 
-        CGSize indsize = [UIProgressIndicator defaultSizeForStyle:kUIProgressIndicatorStyleMediumWhite];
+        CGSize indsize = [UIProgressIndicator defaultSizeForStyle:UIProgressIndicatorStyleMediumWhite];
         indicator_ = [[UIProgressIndicator alloc] initWithFrame:CGRectMake(281, 12, indsize.width, indsize.height)];
-        [indicator_ setStyle:kUIProgressIndicatorStyleMediumWhite];
+        [indicator_ setStyle:UIProgressIndicatorStyleMediumWhite];
 
         Package *package([database_ packageWithName:@"cydia"]);
         NSString *application = package == nil ? @"Cydia" : [NSString
@@ -5619,8 +5605,8 @@
         bool ugly = barstyle == 0;
 
         UIProgressIndicatorStyle style = ugly ?
-            kUIProgressIndicatorStyleMediumBrown :
-            kUIProgressIndicatorStyleMediumWhite;
+            UIProgressIndicatorStyleMediumBrown :
+            UIProgressIndicatorStyleMediumWhite;
 
         CGSize indsize = [UIProgressIndicator defaultSizeForStyle:style];
         unsigned indoffset = (ovrrect.size.height - indsize.height) / 2;
