diff -ru cydia-1.0.2595/cydia/Cydia.mm cydia-1.0.2595+iPhone/cydia/Cydia.mm
--- cydia-1.0.2595/cydia/Cydia.mm	2008-09-23 23:10:25.000000000 +0000
+++ cydia-1.0.2595+iPhone/cydia/Cydia.mm	2008-11-21 17:39:24.000000000 +0000
@@ -6223,7 +6223,7 @@
     CGSize keysize = [UIKeyboard defaultSize];
     CGRect keyrect = {{0, [overlay_ bounds].size.height}, keysize};
     keyboard_ = [[UIKeyboard alloc] initWithFrame:keyrect];
-    [[UIKeyboardImpl sharedInstance] setSoundsEnabled:(Sounds_Keyboard_ ? YES : NO)];
+    //[[UIKeyboardImpl sharedInstance] setSoundsEnabled:(Sounds_Keyboard_ ? YES : NO)];
     [overlay_ addSubview:keyboard_];
 
     install_ = [[InstallView alloc] initWithBook:book_ database:database_];
