diff -ru cydia-1.0.2517/cydia/Cydia.app/package.js cydia-1.0.2517+iPhone/cydia/Cydia.app/package.js
--- cydia-1.0.2517/cydia/Cydia.app/package.js	2008-08-10 13:14:08.000000000 +0000
+++ cydia-1.0.2517+iPhone/cydia/Cydia.app/package.js	2008-08-10 13:34:05.000000000 +0000
@@ -79,7 +79,7 @@
         $("#maintainer-link").href("mailto:" + maintainer.address + "?subject=" + regarding);
     }
 
-    var sponsor = package.maintainer;
+    var sponsor = package.sponsor;
     if (sponsor == null)
         $(".sponsor").remove();
     else {
