diff -ru ncurses-5.6/configure ncurses-5.6+iPhone/configure
--- ncurses-5.6/configure	2006-12-17 16:33:38.000000000 +0000
+++ ncurses-5.6+iPhone/configure	2008-02-23 11:27:37.000000000 +0000
@@ -4922,7 +4922,7 @@
 	darwin*)
 		EXTRA_CFLAGS="-no-cpp-precomp"
 		CC_SHARED_OPTS="-dynamic"
-		MK_SHARED_LIB='${CC} ${CFLAGS} -dynamiclib -install_name ${DESTDIR}${libdir}/`basename $@` -compatibility_version ${ABI_VERSION} -current_version ${ABI_VERSION} -o $@'
+		MK_SHARED_LIB='${CC} ${CFLAGS} -dynamiclib -install_name ${libdir}/`basename $@` -compatibility_version ${ABI_VERSION} -current_version ${ABI_VERSION} -o $@'
 		test "$cf_cv_shlib_version" = auto && cf_cv_shlib_version=abi
 		cf_cv_shlib_version_infix=yes
 		echo "$as_me:4928: checking if ld -search_paths_first works" >&5
