diff -r -u system_cmds-431/shutdown.tproj/shutdown.c system_cmds-431+iPhone/shutdown.tproj/shutdown.c
--- system_cmds-431/shutdown.tproj/shutdown.c	2007-09-06 21:59:26.000000000 +0000
+++ system_cmds-431+iPhone/shutdown.tproj/shutdown.c	2008-02-25 22:30:51.000000000 +0000
@@ -471,7 +471,7 @@
 		if (nosync) howto |= RB_NOSYNC;
 
 		// launchd(8) handles reboot.  This call returns NULL on success.
-		exit(reboot2(howto) == NULL ? EXIT_SUCCESS : EXIT_FAILURE);
+                reboot(howto);
 	}
 	/* NOT-REACHED */
 
