diff -r -ru mobileterminal-286/Settings.h mobileterminal-286+iPhone/Settings.h
--- mobileterminal-286/Settings.h	2008-05-07 08:33:09.000000000 +0000
+++ mobileterminal-286+iPhone/Settings.h	2008-05-07 08:30:30.000000000 +0000
@@ -18,12 +18,13 @@
   NSString * font;
   NSString * args;
   
-  RGBAColor colors[NUM_TERMINAL_COLORS];
+  RGBAColor _colors[NUM_TERMINAL_COLORS];
 }
 
 - (NSString*) fontDescription;
+- (RGBAColor *) colors;
 
-@property RGBAColor * colors;
+@property (getter = colors) RGBAColor *colors;
 @property BOOL autosize;
 @property int width;
 @property int fontSize;
diff -r -ru mobileterminal-286/Settings.m mobileterminal-286+iPhone/Settings.m
--- mobileterminal-286/Settings.m	2008-05-07 08:33:09.000000000 +0000
+++ mobileterminal-286+iPhone/Settings.m	2008-05-07 08:30:20.000000000 +0000
@@ -62,7 +62,7 @@
 }
 
 - (RGBAColor*) colors {
-  return colors;
+  return _colors;
 }
 
 //_______________________________________________________________________________
