diff -ru llvm-gcc-4.2-47984/gcc/llvm-abi.h llvm-gcc-4.2-47984+iPhone/gcc/llvm-abi.h
--- llvm-gcc-4.2-47984/gcc/llvm-abi.h	2008-03-06 02:50:05.000000000 +0000
+++ llvm-gcc-4.2-47984+iPhone/gcc/llvm-abi.h	2008-03-06 12:37:55.000000000 +0000
@@ -206,7 +206,7 @@
 // should be returned using the aggregate shadow (sret) convention, 0 otherwise.
 // X may be evaluated more than once.
 #ifndef LLVM_SHOULD_RETURN_VECTOR_AS_SHADOW
-#define LLVM_SHOULD_RETURN_VECTOR_AS_SHADOW(X) 0
+#define LLVM_SHOULD_RETURN_VECTOR_AS_SHADOW(X,Y) 0
 #endif
 
 /// DefaultABI - This class implements the default LLVM ABI where structures are
