diff -ru llvm-gcc-4.2-47984/gcc/config/darwin.h llvm-gcc-4.2-47984+iPhone/gcc/config/darwin.h
--- llvm-gcc-4.2-47984/gcc/config/darwin.h	2008-02-08 18:58:15.000000000 +0000
+++ llvm-gcc-4.2-47984+iPhone/gcc/config/darwin.h	2008-03-06 12:38:16.000000000 +0000
@@ -1038,6 +1038,7 @@
 /* Macros defining the various PIC cases.  */
 
 /* APPLE LOCAL mdynamic-no-pic */
+#undef MACHO_DYNAMIC_NO_PIC_P
 #define MACHO_DYNAMIC_NO_PIC_P	(TARGET_MACHO_DYNAMIC_NO_PIC)
 /* APPLE LOCAL begin mach-o cleanup */
 #undef MACHOPIC_INDIRECT
