diff -ru coreutils-7.2/Makefile.am coreutils-7.2+iPhone/Makefile.am
--- coreutils-7.2/Makefile.am	2009-03-29 17:44:31.000000000 +0000
+++ coreutils-7.2+iPhone/Makefile.am	2009-04-11 00:05:17.000000000 +0000
@@ -15,7 +15,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-SUBDIRS = lib src doc man po tests gnulib-tests
+SUBDIRS = lib src doc po tests gnulib-tests
 EXTRA_DIST = cfg.mk maint.mk \
   .prev-version THANKS-to-translators THANKStt.in \
   .version \
diff -ru coreutils-7.2/Makefile.in coreutils-7.2+iPhone/Makefile.in
--- coreutils-7.2/Makefile.in	2009-03-31 12:35:42.000000000 +0000
+++ coreutils-7.2+iPhone/Makefile.in	2009-04-11 00:05:23.000000000 +0000
@@ -931,7 +931,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = lib src doc man po tests gnulib-tests
+SUBDIRS = lib src doc po tests gnulib-tests
 EXTRA_DIST = cfg.mk maint.mk \
   .prev-version THANKS-to-translators THANKStt.in \
   .version \
