diff -ru javasqlite-20071108/Makefile.in javasqlite-20071108+iPhone/Makefile.in
--- javasqlite-20071108/Makefile.in	2007-10-25 06:30:12.000000000 +0000
+++ javasqlite-20071108+iPhone/Makefile.in	2008-03-03 06:42:20.000000000 +0000
@@ -122,12 +122,12 @@
 
 libsqlite_jni.la:	sqlite_jni.lo
 	$(LIBTOOL) $(CC) -o libsqlite_jni.la sqlite_jni.lo \
-	    -rpath $(NATIVE_LIB_DIR) $(LIBS)
+	    $(LIBS)
 
 native/mkconst:	native/mkconst.c
-	$(LIBTOOL) $(CC) $(SQLITE_INC) $(SQLITE3_INC) \
+	$(LIBTOOL) gcc $(SQLITE_INC) $(SQLITE3_INC) \
 	    -DHAVE_SQLITE2=@HAVE_SQLITE2@ -DHAVE_SQLITE3=@HAVE_SQLITE3@ \
-	    -o native/mkconst native/mkconst.c $(LIBS)
+	    -o native/mkconst native/mkconst.c
 
 Makefile:	Makefile.in configure config.status
 	./config.status
