Index: lib/Target/ARM/ARMTargetAsmInfo.cpp
===================================================================
--- lib/Target/ARM/ARMTargetAsmInfo.cpp	(revision 47904)
+++ lib/Target/ARM/ARMTargetAsmInfo.cpp	(working copy)
@@ -49,7 +49,7 @@
     PrivateGlobalPrefix = "L";
     BSSSection = 0;                       // no BSS section.
     ZeroFillDirective = "\t.zerofill\t";  // Uses .zerofill
-    SetDirective = "\t.set";
+    SetDirective = "\t.set\t";
     WeakRefDirective = "\t.weak_reference\t";
     HiddenDirective = "\t.private_extern\t";
     ProtectedDirective = NULL;
