diff --git /usr/include/utmp.h /usr/include/utmp.h
index a361c50..24d6f50 100644
--- /usr/include/utmp.h
+++ /usr/include/utmp.h
@@ -92,7 +92,7 @@ struct lastlog {
 	time_t	ll_time;
 	char	ll_line[UT_LINESIZE];
 	char	ll_host[UT_HOSTSIZE];
-}					__OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0,__MAC_10_5,__IPHONE_NA,__IPHONE_NA);
+}					__OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0,__MAC_10_5,__IPHONE_1_2,__IPHONE_2_0);
 
 struct utmp {
 	char	ut_line[UT_LINESIZE];
